process/smartContract/process.go

```diff
@@ -758,6 +775,27 @@ func (sc *scProcessor) processVMOutput(
 		return nil, nil, process.ErrNilTransaction
 	}
 
+	if callType != vmcommon.AsynchronousCall && isTooMuchGasProvided(gasProvided, vmOutput.GasRemaining) {
```

Comment:
duplicated code

Reply:
Might be a separated function for this duplicated code

Reply:
done